### PR TITLE
Extract reusable cancellable file search utility

### LINKED ISSUE #540: Extract reusable cancellable file search utility
## Context

PR #538 introduced an efficient cancellable `fast-glob` traversal for recipe-local discovery.

The implementation currently keeps recipe-specific discovery policy and generic filesystem traversal together in `client/src/documentLinkProvider.ts`.

Following maintainer feedback, the most useful reusable boundary is the filesystem search itself rather than a larger module split.

## Proposed work

Extract a small `CancellableFileSearch` utility responsible for the reusable traversal mechanics:

- run efficient file/folder searches from supplied glob patterns;
- preserve bounded concurrency and the current traversal options;
- observe pre-existing cancellation;
- destroy an active stream when cancellation is requested;
- dispose the cancellation subscription;
- clean up the stream when traversal finishes or another error occurs;
- return the discovered files and directories without introducing recipe-specific policy.

Keep recipe-local responsibilities close to `BitbakeDocumentLinkProvider` for now:

- deciding whether recipe-local discovery applies;
- constructing recipe-local roots;
- producing glob patterns;
- associating discovered URIs with document links.

A separate `RecipeLocalDiscovery` class in the same TypeScript file is acceptable if it improves the distinction without adding unnecessary module structure, but it is not a goal by itself.

## Testing approach

Reuse the functional and edge-case coverage added with #538.

Add or adjust tests only where the extraction introduces project-owned behavior that is not already covered. Do not add tests for `fast-glob` implementation details or duplicate traversal-library guarantees.

## Non-goals

- adding a benchmark or performance harness;
- introducing timing or memory thresholds;
- broadly splitting document-link and recipe-local discovery into separate modules;
- adding exhaustive traversal tests;
- changing completion or document-link semantics;
- introducing a timeout or result truncation.

## Acceptance criteria

- generic cancellable traversal is encapsulated in a reusable `CancellableFileSearch` utility;
- recipe-specific discovery delegates filesystem traversal to that utility;
- existing user-visible behavior remains unchanged;
- cancellation and cleanup behavior from #538 is preserved;
- no benchmark or redundant library-level test coverage is added;
- lint, TypeScript compilation, and relevant existing tests pass.

## Relationship to existing work

This is a small maintainability follow-up to #537 and merged PR #538. It intentionally follows the narrower KISS direction discussed during review.

### PR DESCRIPTION
**Pull request by gcomneno**
## Summary

Extract the generic cancellable filesystem traversal from `BitbakeDocumentLinkProvider` into a reusable `CancellableFileSearch` utility.

Recipe-local policy remains close to `BitbakeDocumentLinkProvider`; only the generic search mechanics move to the utility. The language client now uses the same reusable search directly.

The refactor preserves the behavior introduced by #538, including:

- bounded `fast-glob` traversal;
- pre-existing cancellation handling;
- active stream destruction on cancellation;
- cancellation subscription disposal;
- stream cleanup;
- file-result limiting.

No benchmark, timeout, result truncation, or additional traversal-library test coverage is introduced.

## Validation

- document-link provider tests: 4 passed
- server unit tests: 86 passed
- TypeScript compilation passed
- ESLint passed
- `git diff --check` passed

Generated parser, documentation, and SPDX test resources were prepared using the repository-provided fetch scripts and remained ignored by Git.

Closes #540

### CODE REVIEW

File: client/src/documentLinkProvider.ts [hunk 1]
```diff
@@ -6,21 +6,15 @@
 import vscode from 'vscode'
 import { type LanguageClient } from 'vscode-languageclient/node'
 import { RequestMethod, type RequestResult } from './lib/src/types/requests'
-import { logger } from './lib/src/utils/OutputLogger'
 import path from 'path'
 import { extractRecipeName } from './lib/src/utils/files'
+import { CancellableFileSearch } from './utils/CancellableFileSearch'
 import fg from 'fast-glob'
```
Comment (deribaucourt) [inline client/src/documentLinkProvider.ts:12]: We likely no longer need fast-glob in this file anymore thanks to the refactor. Maybe path as well.
Make sure to cleanup the imports.

**Review by deribaucourt — CHANGES_REQUESTED** [review]
Looks good, one minor remark

File: client/src/documentLinkProvider.ts (hunk 1 above)
Comment (gcomneno) [reply client/src/documentLinkProvider.ts:12]: Thanks for pointing this out. I checked both imports. `fast-glob` is still used here for `convertPathToPattern()` and `escapePath()` when building the recipe-local patterns; only the traversal via `fg.stream()` moved to `CancellableFileSearch`. `path` is also still used for `extname`, `dirname`, `join`, and `basename`. So both imports are still required.

**Review by deribaucourt — APPROVED** [review]
